vm/vm.go

```diff
@@ -805,9 +811,7 @@ func (vm *VM) GetValidityWindow() *chain.TimeValidityWindow {
 	return vm.chainTimeValidityWindow
 }
 
-// implements "block.ChainVM.commom.VM.Parser"
-// replaces "core.SnowmanVM.ParseBlock"
-func (vm *VM) ParseBlock(ctx context.Context, source []byte) (snowman.Block, error) {
+func (vm *VM) ParseStatefulBlock(ctx context.Context, source []byte) (*StatefulBlock, error) {
```

Comment:
Why did we rename this? Wondering if we need this + the `ParseBlock` function... I guess this one is nice because it returns the concrete type but it's also not something we're using right now.